mmv1/third_party/terraform/tests/resource_sql_user_test.go

```diff
@@ -279,6 +279,43 @@ func testAccSqlUserDestroyProducer(t *testing.T) func(s *terraform.State) error
 	}
 }
 
+func TestAccSqlUser_mysqlPasswordPolicy(t *testing.T) {
```

Comment:
Were you able to successfully run this test?

I would be very surprised if this works due to the lack of a flatten method for the new block

Reply:
The tests are passing fine, This resource does not have a flatten method for any block. I have followed the same coding style.

Reply:
Ah, tests pass fine but this doesn't quite work as we would like it to. Currently the password_policy field is untracked, meaning this resource doesn't read the password_policy field from the API and compare it to state, it just assumes the value in state is present in the API. This makes sense for fields that aren't returned by the API like `password` itself, but doesn't match our normal pattern

We should track it as it does come back in the API response.

Reply:
Understood, thanks for explaining, have fixed this and tested the code.